src/main/java/org/jabref/gui/actions/ExportLinkedFilesAction.java

```diff
@@ -0,0 +1,143 @@
+
+        DirectoryDialogConfiguration dirDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                .withInitialDirectory(Paths.get(Globals.prefs.get(JabRefPreferences.EXPORT_WORKING_DIRECTORY)))
+                .build();
+        entries = JabRefGUI.getMainFrame().getCurrentBasePanel().getSelectedEntries();
+
+        Optional<Path> exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> dialogService.showDirectorySelectionDialog(dirDialogConfiguration));
+
+        exportPath.ifPresent(path -> {
+            databaseContext = JabRefGUI.getMainFrame().getCurrentBasePanel().getDatabaseContext();
+
+            Service<List<CopyFilesResultViewModel>> exportService = new ExportLinkedFilesService(databaseContext, entries, path);
+            startServiceAndshowProgessDialog(exportService);
+        });
+    }
+
+    private void startServiceAndshowProgessDialog(Service<List<CopyFilesResultViewModel>> exportService) {
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+            exportService.setOnSucceeded(value -> {
+                DefaultTaskExecutor.runInJavaFXThread(() -> showDialog(exportService.getValue()));
+            });
+            exportService.start();
+            DialogService dlgService = new FXDialogService();
+            dlgService.showCanceableProgressDialogAndWait(exportService);
+
+        });
+    }
+
+    private void showDialog(List<CopyFilesResultViewModel> data) {
+        Dialog<ButtonType> dlg = new Dialog<>();
+        dlg.setTitle(Localization.lang("Result"));
+        ObservableList<CopyFilesResultViewModel> tableData = FXCollections.observableArrayList(data);
+
+        TableView<CopyFilesResultViewModel> tv = createTable();
+        ScrollPane sp = new ScrollPane();
+        sp.setContent(tv);
+
+        dlg.getDialogPane().setContent(sp);
+        tv.setItems(tableData);
+
+        sp.setFitToHeight(true);
+        sp.setFitToWidth(true);
+
+        dlg.getDialogPane().getButtonTypes().add(ButtonType.CLOSE);
+        dlg.setResizable(true);
+        dlg.showAndWait();
+    }
+
+    private static TableView<CopyFilesResultViewModel> createTable() {
+        TableView<CopyFilesResultViewModel> tableResult = new TableView<>();
+
+        TableColumn<CopyFilesResultViewModel, String> colFile = new TableColumn<>(Localization.lang("File"));
+        TableColumn<CopyFilesResultViewModel, MaterialDesignIcon> colIcon = new TableColumn<>(Localization.lang("Status"));
+        TableColumn<CopyFilesResultViewModel, String> colMessage = new TableColumn<>(Localization.lang("Message"));
+
+        colFile.setCellValueFactory(cellData -> cellData.getValue().getFile());
+        colMessage.setCellValueFactory(cellData -> cellData.getValue().getMessage());
+        colIcon.setCellValueFactory(cellData -> cellData.getValue().getIcon());
+
+        colIcon.setCellFactory(column -> {
+            return new TableCell<CopyFilesResultViewModel, MaterialDesignIcon>() {
```

Comment:
I think there is a table cell factory helper (similar to list view). If not, then I would be nice if you could create one.